app/logic/jwt_authenticator.rb

```diff
@@ -41,6 +50,9 @@ def token_from_params(params)
     def store_token_in_cookie(token, cookies)
       cookie = { TokenAuthenticationController::JWT_TOKEN_COOKIE => token }
       CookieJar.set_or_update_cookie(cookies, cookie)
+      # Remove the TokenAuthenticator cookie if set, so that when the token is no
+      # longer in the params, we still end up with the JWT token login.
+      CookieJar.delete_cookie(cookies, TokenAuthenticationController::PERSON_ID_COOKIE)
```

Comment:
We could, but we don't have to:
![rubymine64_2020-02-20_16-04-19](https://user-images.githubusercontent.com/607965/74947107-a7c73080-53fa-11ea-8452-bfcd7bb7e1ca.png)

In the order above, it always tries to read the cookie set by the questionnaire-url-in-sms (`TokenAuthenticator`) first, before trying to read the cookie stored from giving an `auth` or `token` query param in an earlier request (`JwtAuthenticator`).

However, I agree that it would be a more clean solution to say that if you're logged in as one person, you can't go back to the session of another person, simply by deleting one cookie. That would be confusing for people who thought they were logged out by the fact that another person logged in on the same computer.

I'll fix it.

Reply:
Fixed.